qiskit/extensions/qasm_simulator_cpp/load.py

```diff
@@ -17,63 +16,82 @@
 # =============================================================================
 
 """
-local_qiskit_simulator command to load a saved quantum state.
+Simulator command to load a saved quantum state.
 """
-from qiskit import CompositeGate
-from qiskit import Gate
+from qiskit import Instruction
 from qiskit import QuantumCircuit
-from qiskit._instructionset import InstructionSet
-from qiskit._quantumregister import QuantumRegister
-from qiskit.qasm import _node as node
+from qiskit import CompositeGate
+from qiskit import QuantumRegister
+from qiskit.extensions._extensionerror import ExtensionError
+from qiskit.extensions.standard import header  # pylint: disable=unused-import
+
 
+class Load(Instruction):
+    """Simulator load instruction."""
 
-class LoadGate(Gate):
-    """Simulator load operation."""
+    def __init__(self, slot, qubits, circ):
+        """Create new load instruction."""
+        super().__init__("load", [slot], list(qubits), circ)
 
-    def __init__(self, m, qubit, circ=None):
-        """Create new load gate."""
-        super().__init__("load", [m], [qubit], circ)
+    def inverse(self):
+        """Special case. Return self."""
+        return self
 
     def qasm(self):
         """Return OPENQASM string."""
-        qubit = self.arg[0]
-        m = self.param[0]
-        return self._qasmif("load(%d) %s[%d];" % (m, qubit[0].name,
-                                                  qubit[1]))
-
-    def inverse(self):
-        """Invert this gate."""
-        return self  # self-inverse
+        string = "load(%d) " % self.param[0]
+        for j in range(len(self.arg)):
+            if len(self.arg[j]) == 1:
+                string += "%s" % self.arg[j].openqasm_name
+            else:
+                string += "%s[%d]" % (self.arg[j][0].openqasm_name, self.arg[j][1])
+            if j != len(self.arg) - 1:
+                string += ","
+        string += ";"
+        return string
 
     def reapply(self, circ):
-        """Reapply this gate to corresponding qubits in circ."""
-        self._modifiers(circ.load(self.param[0], self.arg[0]))
+        """Reapply this instruction to corresponding qubits in circ."""
+        self._modifiers(circ.snapshot(self.param[0], *self.arg))
```

Comment:
Can you double-check if this is meant to be `circ.load(...)`?

Reply:
This should be `circ.load`

Reply:
good catch.